### PR TITLE
Scroll width tracking

### PR DESCRIPTION
**Pull request by wise0704**
http://www.flashdevelop.org/community/viewtopic.php?f=5&t=12589

Setting `ScrollWidth` to `0` should enable `ScrollWidthTracking` too.

### CODE REVIEW

File: FlashDevelop/Managers/ScintillaManager.cs [hunk 1]
```diff
@@ -253,7 +253,8 @@ public static void ApplySciSettings(ScintillaControl sci, Boolean hardUpdate)
                 sci.IsTabIndents = settings.TabIndents;
                 sci.IsUseTabs = settings.UseTabs;
                 sci.IsViewEOL = settings.ViewEOL;
-                sci.ScrollWidth = settings.ScrollWidth;
+                sci.ScrollWidth = Math.Max(settings.ScrollWidth, 1);
```
Comment (Meychi) [inline FlashDevelop/Managers/ScintillaManager.cs]: Is 1 minimum width?

File: FlashDevelop/Settings/Accessors.cs [hunk 2]
```diff
@@ -294,7 +294,11 @@ public Int32 CaretWidth
         public Int32 ScrollWidth
         {
             get { return this.scrollWidth; }
-            set { this.scrollWidth = value; }
+            set
+            {
+                if (value < 0) throw new ArgumentOutOfRangeException();
```
Comment (Meychi) [inline FlashDevelop/Settings/Accessors.cs]: I don't think throwing error here is good. Maybe just reset it to 0 if its negative.

Comment (Neverbirth) [inline FlashDevelop/Settings/Accessors.cs]: I think throwing an error is the standard thing to do in this case.

File: FlashDevelop/Managers/ScintillaManager.cs (hunk 1 above)
Comment (wise0704) [inline FlashDevelop/Managers/ScintillaManager.cs]: Yep, setting it to 0 sets it to the scintilla default (which is I think either 2000 or 5000).

File: FlashDevelop/Settings/Accessors.cs (hunk 2 above)
Comment (wise0704) [inline FlashDevelop/Settings/Accessors.cs]: what should i do...

**Comment by Meychi** [comment]
Well match it to other similar settings. Try put an invalid value and check if it gives an error or not.

**Comment by wise0704** [comment]
@Meychi 
# ???????????
image

**Comment by wise0704** [comment]
That I'm sure isn't expected...

**Comment by wise0704** [comment]
`PropertyGrid` by default does this however:
image

**Pull request merged**